ui/src/app/base/components/NotificationGroup/NotificationGroup.tsx

```diff
@@ -0,0 +1,107 @@
+import classNames from "classnames";
+import pluralize from "pluralize";
+import PropTypes from "prop-types";
+import React from "react";
+import { useDispatch } from "react-redux";
+import { Button, Notification } from "@canonical/react-components";
+import type { Dispatch } from "redux";
+
+import { capitaliseFirst } from "app/utils";
+import { useVisible } from "app/base/hooks";
+import { notification as notificationActions } from "app/base/actions";
+import type { Notification as NotificationType } from "app/store/notification/types";
+
+export const notificationTypes = {
+  CAUTION: "caution",
+  INFORMATION: "information",
+  NEGATIVE: "negative",
+  POSITIVE: "positive",
+};
+
+const dismissAll = (notifications: NotificationType[], dispatch: Dispatch) => {
+  notifications.forEach((notification) => {
+    if (notification.dismissable) {
+      dismiss(notification.id, dispatch);
+    }
+  });
+};
+
+const dismiss = (id: NotificationType["id"], dispatch: Dispatch) => {
+  dispatch(notificationActions.delete(id));
+};
+
+type Props = {
+  notifications: NotificationType[];
+  type: "caution" | "negative" | "positive" | "information";
```

Comment:
We have a type for this in `types`, `MessageType`.

Reply:
So we do... thanks!